File: tests/inputs/hello.swift
// Sample Swift file to exercise syntax highlighting, indents,
// and textobjects in vorto. Open with `vorto assets/samples/hello.swift`.

import Foundation

/// A tiny program exercising typical Swift constructs:
/// structs, enums with associated values, protocols, generics,
/// guards, optionals, closures, and string interpolation.

protocol Greeter {
    associatedtype Subject
    func greet(_ subject: Subject) -> String
}

struct Person: Equatable {
    let name: String
    var age: Int

    var isAdult: Bool { age >= 18 }
}

enum Classification: CustomStringConvertible {
    case negative
    case zero
    case positive(even: Bool)

    var description: String {
        switch self {
        case .negative: return "negative"
        case .zero: return "zero"
        case .positive(let even): return even ? "positive even" : "positive odd"
        }
    }
}

struct PersonGreeter: Greeter {
    var prefix: String = "Hello"

    func greet(_ subject: Person) -> String {
        "\(prefix), \(subject.name)!"
    }
}

func classify(_ n: Int) -> Classification {
    guard n != 0 else { return .zero }
    if n < 0 { return .negative }
    return .positive(even: n.isMultiple(of: 2))
}

func summarize<T: Sequence>(_ xs: T) -> String where T.Element == Int {
    let mapped = xs.map { $0 * $0 }.filter { $0 > 4 }
    return mapped.map(String.init).joined(separator: ", ")
}

let people = [
    Person(name: "Alice", age: 30),
    Person(name: "Bob", age: 17),
    Person(name: "Carol", age: 42),
]

let greeter = PersonGreeter(prefix: "Hi")
for p in people where p.isAdult {
    print(greeter.greet(p))
}

for n in -2...5 {
    print("\(n) is \(classify(n))")
}

print("squares > 4: \(summarize(1...6))")

File: tests/inputs/hello.ex
# Sample Elixir file to exercise syntax highlighting, indents,
# and textobjects in vorto. Open with `vorto assets/samples/hello.ex`.

defmodule Greeter do
  @moduledoc """
  Tiny greeter module to show typical Elixir constructs:
  module attributes, structs, pattern matching, pipes,
  guards, sigils, and comprehensions.
  """

  @default_greeting "Hello"

  defstruct [:name, age: 0, tags: []]

  @typedoc "A person with a name, age, and free-form tags."
  @type t :: %__MODULE__{name: String.t(), age: non_neg_integer(), tags: [atom()]}

  @doc "Greet a person, optionally with a custom prefix."
  @spec greet(t(), String.t()) :: String.t()
  def greet(%__MODULE__{name: name}, prefix \\ @default_greeting)
      when is_binary(prefix) and prefix != "" do
    "#{prefix}, #{name}!"
  end

  def classify(n) when is_integer(n) do
    cond do
      n < 0 -> :negative
      n == 0 -> :zero
      rem(n, 2) == 0 -> :positive_even
      true -> :positive_odd
    end
  end

  def squares(range \\ 1..10) do
    range
    |> Enum.map(&(&1 * &1))
    |> Enum.filter(fn n -> n > 5 end)
    |> Enum.with_index()
  end

  def tag_words do
    for word <- ~w(alpha beta gamma), into: %{} do
      {word, String.length(word)}
    end
  end
end

defprotocol Shoutable do
  @doc "Convert a thing to a SHOUTING string."
  def shout(value)
end

defimpl Shoutable, for: BitString do
  def shout(s), do: String.upcase(s) <> "!!!"
end

# Script entry — runs when executed with `elixir hello.ex`.
alice = %Greeter{name: "Alice", age: 30, tags: [:admin, :early_bird]}
IO.puts(Greeter.greet(alice))
IO.puts(Greeter.greet(alice, "Hi"))
IO.inspect(Greeter.classify(-3))
IO.inspect(Greeter.squares())
IO.inspect(Greeter.tag_words())
IO.puts(Shoutable.shout("hello"))
